apps/core/lib/core/event_repo.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Core.EventRepo do
+  @type operation ::
+          :user_created
+          | :user_updated
+          | :user_deleted
+          | :post_created
+
+@spec save_event(operation, {pos_integer(), String.t(), any()}) :: :ok | {:error, any()}
+  def save_event(:user_created, {user_id, user_email, time}),
+    do: write_to_mnesia(user_id, :user_created, %{user_email: user_email}, time)
+
+  def save_event(:user_updated, {user_id, attrs, time}),
+    do: write_to_mnesia(user_id, :user_updated, attrs, time)
+
+  def save_event(:user_deleted, {user_id, time}),
+    do: write_to_mnesia(user_id, :user_deleted, %{}, time)
+
+  def save_event(:post_created, {user_id, post_id, post_body, time}),
+    do:
```

Comment:
Since this definition is multiline I would use the macro syntax (`def .. do ... end`).

Reply:
mix format splitted the lines ;)
But if we're just calling other function shouldn't `def .. , do: ... ` syntax be good as well? I see no point in adding macro syntax just because formetter splitted argument list, but I may be wrong here